content/blog/2020-01-ggplot2-3-3-0/index.Rmarkdown

```diff
@@ -0,0 +1,231 @@
+```
+
+### More control over aesthetic evaluation
+When mapping data to aesthetics we are usually differentiating between two different mappings. One for the stat and one for the geom. We don't think about this that often, because the stat simply accept the mappings and carry them over to the geom (sometimes with modifications). Sometimes, though, we want to control the mapping that happens between the output of the stat and the geom. The approach to this has been to use `stat()` (and `..var..` notation before that), to indicate that the aesthetic should be evaluated after the stat has been computed.
+
+With this release we expand on this, as well as making it more explicit. There are now three entry points to evaluation: *start*, *after_stat*, and *after_scale*. *start* is the default (i.e. evaluated in the context of the layer data), *after_stat* is like the old `stat()` (i.e. evaluated in the context of the stat output), and *after_scale* is new and specifies that evaluation should occur after the aesthetic has been scaled. The `after_stat()` and `after_scale()` functions are used to mark aesthetics for their respective delayed evaluation (thus soft-deprecating `stat()`):
+
+You would use `after_stat()` to e.g. pick a different computed metric than the default (here _density_ instead of _count_)
+
+```{r, message=FALSE}
+ggplot(mpg, aes(displ)) +
+  geom_histogram(aes(y = after_stat(density)))
+```
+
+And you can use `after_scale()` to assign fill as a variant of the scaled color:
+
+```{r}
+ggplot(mpg, aes(class, hwy)) +
+  geom_boxplot(aes(colour = class, fill = after_scale(alpha(colour, 0.4))))
+```
+
+Furthermore, it is now possible to perform multiple mapping for the same aesthetic, using the `stage()` function. This can be used to e.g. set alpha on the stroke as the default is to only apply alpha to fill for polygon-type geoms:
+
+```{r}
+ggplot(mpg) + 
+  geom_bar(
+    aes(
+      x = drv, 
+      colour = stage(start = drv, after_scale = alpha(colour, 0.5))
+    ), 
+    fill = NA, size = 4
+  )
+```
+
+`stage()` is definitely not something you need every day, but it does resolve a range of issues (like the one above) and it is nice to have full transparency and control over the aesthetic evaluation.
+
+### More extensible theming
+The possibility to extend coordinate systems and faceting has created a need to allow new theme elements to be specified (along with inheritance). One could envision e.g. a faceting setup that related panels with arrows, and in order to control the look of the arrows we would need to allow a call like `theme(panel.arrow = element_line(...))`. With this release we finally provide a way for extension developers to register new theme elements using `register_theme_elements()`. We urge extension developers to namespace the theme elements they create in order to avoid name clashes. If the above facet was part of the ggrelfacet package the registration would look something like this:
```

Comment:
```suggestion
The possibility to extend coordinate systems and faceting has created a need to allow new theme elements to be specified (along with inheritance). One could envision e.g. a faceting setup that related panels with arrows, and in order to control the look of the arrows we would need to allow a call like `theme(panel.arrow = element_line(...))`. With this release, we finally provide a way for extension developers to register new theme elements using `register_theme_elements()`. We urge extension developers to namespace the theme elements they create in order to avoid name clashes. If the above facet was part of the ggrelfacet package the registration would look something like this:
```

Reply:
Added a comma. I'd also suggest adding a note about there being changes particularly relevant to extension developers in the opening paragraph.